rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/server/RMServiceImpl.java

```diff
@@ -288,222 +258,146 @@ public String defineNodeSource(final String sessionId, final String nodeSourceNa
-            final List<String> attrs) throws RestServerException, ServiceException {
-        return executeFunctionReturnStreamAsString(new Function<RestClient, InputStream>() {
-            @Override
-            public InputStream apply(RestClient restClient) {
-                return restClient.getNodeMBeanInfo(sessionId, nodeJmxUrl, objectName, attrs);
-            }
-        });
+    public String getNodeMBeanInfo(String sessionId, String nodeJmxUrl, String objectName, List<String> attrs)
+            throws RestServerException, ServiceException {
+        return executeFunctionReturnStreamAsString(restClient -> restClient.getNodeMBeanInfo(sessionId,
+                                                                                             nodeJmxUrl,
+                                                                                             objectName,
+                                                                                             attrs));
     }
 
     @Override
-    public String getNodeMBeanHistory(final String sessionId, final String nodeJmxUrl, final String objectName,
-            final List<String> attrs, final String timeRange) throws RestServerException, ServiceException {
-        return executeFunctionReturnStreamAsString(new Function<RestClient, InputStream>() {
-            @Override
-            public InputStream apply(RestClient restClient) {
-                return restClient.getNodeMBeanHistory(sessionId, nodeJmxUrl, objectName, attrs, timeRange);
-            }
-        });
+    public String getNodeMBeanHistory(String sessionId, String nodeJmxUrl, String objectName, List<String> attrs,
+            String timeRange) throws RestServerException, ServiceException {
+        return executeFunctionReturnStreamAsString(restClient -> restClient.getNodeMBeanHistory(sessionId,
+                                                                                                nodeJmxUrl,
+                                                                                                objectName,
+                                                                                                attrs,
+                                                                                                timeRange));
     }
 
     @Override
-    public String getNodeMBeansInfo(final String sessionId, final String nodeJmxUrl, final String objectNames,
-            final List<String> attrs) throws RestServerException, ServiceException {
-        return executeFunctionReturnStreamAsString(new Function<RestClient, InputStream>() {
-            @Override
-            public InputStream apply(RestClient restClient) {
-                return restClient.getNodeMBeansInfo(sessionId, nodeJmxUrl, objectNames, attrs);
-            }
-        });
+    public String getNodeMBeansInfo(String sessionId, String nodeJmxUrl, String objectNames, List<String> attrs)
+            throws RestServerException, ServiceException {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Make this anonymous inner class a lambda [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1604)